lib/render/editor/editor.js

```diff
@@ -2,14 +2,18 @@ $(document).ready(function () {
   $(".mu-kids-submit-button").html("<kids-submit-button></kids-submit-button>");
   $(".mu-kids-reset-button").html("<kids-reset-button></kids-reset-button>");
 
-  Blockly.WorkspaceAudio.prototype.preload = function () {
-    for (var soundName in this.SOUNDS_) {
-      var sound = this.SOUNDS_[soundName];
-      sound.volume = 0.01;
-      var playPromise = sound.play();
-      playPromise && playPromise.then(sound.pause.bind(sound));
-      if (goog.userAgent.IPAD || goog.userAgent.IPHONE) break;
-    }
+  var setBlocklySounds = function () {
+    if (typeof Blockly === 'undefined') return setTimeout(setBlocklySounds, 50);
```

Comment:
Nice! But can't this lead to an infinite loop?

Reply:
Technically speaking, yes.
But:
1) We are already doing something similar in editor.html. 
2) If Blockly never gets defined, the loop is the least of our problems :stuck_out_tongue:

Reply:
Sound reasonable. However, if we are doing this already, this means we are duplicating logic :stuck_out_tongue: . Can we implement a `waitForBlockly(aFunction)` that does actually perform this glue code?

Reply:
I mean: 

```javascript
function waitForBlockly(aFunction) {
   return function () {
       if (typeof Blockly === 'undefined') return setTimeout(aFunction, 50);
       aFunction();
   }
}
```

Reply:
@flbulgarelli Apparently the decision to put setBlocklySounds in editor.js was arbitrary, so i moved it to editor.html in order to avoid code repetition.